includes/data-port/class-sensei-import-csv-reader.php

```diff
@@ -60,21 +60,94 @@ class Sensei_Import_CSV_Reader {
 	public function __construct( $csv_file, $completed_lines = 0, $lines_per_batch = 30 ) {
 		$this->file = new SplFileObject( $csv_file );
 		$this->file->setFlags( SplFileObject::READ_CSV );
+		$this->detect_delimiter();
 
+		$this->file->seek( PHP_INT_MAX );
+		$this->total_lines     = $this->file->key();
+		$this->completed_lines = $completed_lines;
+		$this->is_completed    = $this->completed_lines >= $this->total_lines;
+		$this->lines_per_batch = $lines_per_batch;
+	}
+
+	/**
+	 * Set the delimiter to read the CSV file.
+	 *
+	 * The delimiter detection works testing the delimiter which find more columns.
+	 */
+	private function detect_delimiter() {
 		/**
-		 * Change the default CSV delimiter.
+		 * Filters the default CSV delimiter.
 		 *
 		 * @since 3.2.0
+		 * @since 3.3.0 Updated the default to `false`, so it'll get through the delimiter detection.
+		 * @hook sensei_import_csv_delimiter
 		 *
-		 * @param string  $delimiter The CSV file delimiter.
+		 * @param {string} $delimiter The CSV file delimiter.
 		 */
-		$this->file->setCsvControl( apply_filters( 'sensei_import_csv_delimiter', ',' ) );
+		$forced_delimiter = apply_filters( 'sensei_import_csv_delimiter', false );
```

Comment:
Just noticed that this wasn't picked up by the hook docs. Not sure why. 🤔

Reply:
Humm. Maybe an issue related to multiple `@since`?

Reply:
PR with a fix here: https://github.com/Automattic/sensei/pull/3468
The current configurations are getting files only from the `includes` folder. There is an option that makes it recursive: `"recurse": true`